contracts/starknet/space/space.cairo

```diff
@@ -52,56 +64,124 @@ end
 @event
 func proposal_created(
         proposal_id : felt, proposer_address : EthAddress, proposal : Proposal,
-        metadata_uri_len : felt, metadata_uri : felt*):
+        metadata_uri_len : felt, metadata_uri : felt*, execution_params_len : felt,
+        execution_params : felt*):
 end
 
 @event
 func vote_created(proposal_id : felt, voter_address : EthAddress, vote : Vote):
 end
 
+# Internal utility function to hash data
+func hash_pedersen{pedersen_ptr : HashBuiltin*}(calldata_len : felt, calldata : felt*) -> (
```

Comment:
The starware and starknet.js methods for hashing an array append the length of the array to the end of itself before hashing. Not sure why they did it that way (was debugging for 3 hours trying to figure out why the hashes were not matching lol), but we should do this too so the hash can be easily computed offchain.

Reply:
The way I saw it, we would be giving this to `calldata` (meaning actual calldata would be `calldata + calldata_len`). This reduces computation on-chain, at the cost of adding a little bit of complexity off-chain (the hash data provided to starkware.py will not be the same one submitted to the contract). Wdyt?

Reply:
The way I saw it, we would be giving this to `calldata` (meaning actual calldata would be `calldata + calldata_len`). This reduces computation on-chain, at the cost of adding a little bit of complexity off-chain (the hash data provided to starkware.py will not be the same one submitted to the contract). Wdyt?

Reply:
The way I saw it, we would be giving this to `calldata` (meaning actual calldata would be `calldata + calldata_len`). This reduces computation on-chain, at the cost of adding a little bit of complexity off-chain (the hash data provided to starkware.py will not be the same one submitted to the contract). Wdyt?

Reply:
The way I saw it, we would be giving this to `calldata` (meaning actual calldata would be `calldata + calldata_len`). This reduces computation on-chain, at the cost of adding a little bit of complexity off-chain (the hash data provided to starkware.py will not be the same one submitted to the contract). Wdyt?

Reply:
oh okay I see what you mean. Yeah happy to do it this way. Maybe add a comment where its used to so its super clear https://github.com/snapshot-labs/sx-core/blob/2e6deb8ea8da9cd42a005c229136474c2d25c886/contracts/starknet/space/space.cairo#L289

Reply:
good idea! :) 37783bc73643bb0cf17a3fcad9eef5704849bfef